tests/backend/test_tf_ops.py

```diff
@@ -51,3 +51,26 @@ def test_reduced_density_matrix_fock_states(self, num_modes, setup_backend, cuto
         expected = tf.constant(expected)
 
         assert np.allclose(reduced_dm, expected, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("modes", [[0], [1], [2], [0, 1], [0, 2], [1, 2], [0, 1, 2]])
+    def test_reduced_density_matrix_multiple_modes(self, setup_backend, cutoff, modes, tol):
+        """Test the reduced_density_matrix returns the correct reduced density matrices
+        when tracing out more than one mode."""
+        zero_photon_state = np.zeros([cutoff])
+        zero_photon_state[0] = 1.
+        one_photon_state = np.zeros([cutoff])
+        one_photon_state[1] = 1.
+
+        # create a single-photon state in the second mode
+        state = np.outer(zero_photon_state, one_photon_state)
+        state = np.multiply.outer(state, zero_photon_state)
+        state_dm = mixed(state)
+        state_dm = tf.constant(state_dm)
+
+        reduced_dm = reduced_density_matrix(state_dm, modes, state_is_pure=False)
+
+        alpha_str = np.array(["ab", "cd", "ef"])
+        expected = np.einsum(f"abcdef->{''.join(alpha_str[modes])}", state_dm)
```

Comment:
blew my mind how you put a string inside a string here :grinning:

Reply:
Would be good to use `tf.einsum` here since these are tensorflow objects

Reply:
Is this `einsum` correct? For a reduced density matrix on, say, mode 0, I would expect the formula to be
```
abccdd->ab
```
i.e., we sum the diagonals explicitly. My understanding of einsum is that when you use `->` any index missing from the left of `->` is summed over, but your example would sum over both the row and column indices of the traced out subsystems independently, not the diagonal, as is needed for partial trace. Please can you confirm? (likely this specific example masks it because there is only one nonzero entry)

Reply:
I rewrote the test completely, causing `tf.einsum` not to work since it doesn't support the same axis appearing several times in the einstring (unless it's a trace).

Reply:
Oh, just saw your second reply. Hadn't reloaded the page. :laughing: 

You're completely right @co9olguy. It was a quite bad mistake there. Thanks for spotting it! :1st_place_medal: I rewrote the test, creating instead GHZ state and using that for the calculations. I think I got a bit confused with what was happening earlier as well.

Reply:
That's a good point. They still don't support that yet? If not, we'll have to indeed use `np.einsum` then